ksql-engine/src/test/resources/query-validation-tests/more-complex-struct.json

```diff
@@ -122,31 +122,31 @@
         {
           "topic": "S2",
           "key": 0,
-          "value": "{\"SL4_DOUBLE_ARRAY\":[44.00196898385005,20.811629846775816,18.106233954153627],\"SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_32\",\"SL2_3_DOUBLE\":62.714783363909}}}",
+          "value": "{\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY\":[44.00196898385005,20.811629846775816,18.106233954153627],\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_32\",\"SL2_3_DOUBLE\":62.714783363909}}}",
           "timestamp": 0
         },
         {
           "topic": "S2",
           "key": 100,
-          "value": "{\"SL4_DOUBLE_ARRAY\":[14.483946419811055,48.09062881637733,65.07562628320905],\"SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_60\",\"SL2_3_DOUBLE\":17.481981844681428}}}",
+          "value": "{\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY\":[14.483946419811055,48.09062881637733,65.07562628320905],\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_60\",\"SL2_3_DOUBLE\":17.481981844681428}}}",
           "timestamp": 0
         },
         {
           "topic": "S2",
           "key": 101,
-          "value": "{\"SL4_DOUBLE_ARRAY\":[25.385083987245714,86.69430464168514,17.705377157838733],\"SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_68\",\"SL2_3_DOUBLE\":39.07228201310673}}}",
+          "value": "{\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY\":[25.385083987245714,86.69430464168514,17.705377157838733],\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_68\",\"SL2_3_DOUBLE\":39.07228201310673}}}",
           "timestamp": 0
         },
         {
           "topic": "S2",
           "key": 101,
-          "value": "{\"SL4_DOUBLE_ARRAY\":[74.68834760950283,73.40355939056849,72.15630069159221],\"SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_95\",\"SL2_3_DOUBLE\":51.92447078198864}}}",
+          "value": "{\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY\":[74.68834760950283,73.40355939056849,72.15630069159221],\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_95\",\"SL2_3_DOUBLE\":51.92447078198864}}}",
           "timestamp": 0
         },
         {
           "topic": "S2",
           "key": 101,
-          "value": "{\"SL4_DOUBLE_ARRAY\":[32.27711275538546,3.7338613605496174,52.41743778835066],\"SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_1\",\"SL2_3_DOUBLE\":85.05450978718197}}}",
+          "value": "{\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY\":[32.27711275538546,3.7338613605496174,52.41743778835066],\"NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT_MAP\":{\"mapkey\":{\"SL2_3_STRING\":\"SL2_3_1\",\"SL2_3_DOUBLE\":85.05450978718197}}}",
```

Comment:
Why the change in column names? These new names are pretty horrid. I definitely preferred the previous names.

Reply:
You can easily have conflict in the previous names. In this form you if you select a struct field and don't specify an alias you will have a name created representing the full path of the field. If we used the previous aliasing we would not be able to run valid queries where struct field name is the same as a column name or another struct field name.
Consider the following:

```CREATE STREAM foo (ID BIGINT, ITEM_INFO STRUCT<ID BIGINT>) WITH ...```
```CREATE STREAM bar AS SELECT ID, ITEM_INFO->ID FROM foo;```

Previously the above query would fails since the same alias will be used for two columns, `ID`. But now we will use two aliases: `ID, ITEM_INFO_ID`, and the query won't fail.

Reply:
Yer, but on the down side, `NESTED_STRUCT_FIELD__SL1_STRUCT__SL2_STRUCT__SL3_STRUCT__SL4_DOUBLE_ARRAY` is a bloody horrible column name to be outputting ;) . I guess they can use aliasing to be make it more manageable thought...